src/app/core/filter/filter/filter.component.spec.ts

```diff
@@ -98,6 +98,23 @@ describe("FilterComponent", () => {
     expect(component.filterSelections[0].selectedOptionValues[1]).toBe("bar");
   });
 
+  it("should fail if the URL length exceeds the safe limit", async () => {
```

Comment:
this test is somewhat pointless, because it didn't fail before your code changes either.

Can you work on a better unit test? I would make a call to `filterOptionSelected` in the test action, simulating a loooot of generated selected options (to ensure it'll be >2000 characters). Then spy on the router.navigate call and expect (a) that the url is shorter than 2000 and (b) that the longest param was removed but others stay in the url

Reply:
updated the test cases can you please check now